File: sdk/kotlin/src/main/kotlin/vn/hitechcloud/sdk/Affiliate.kt
package vn.hitechcloud.sdk

/** Affiliate: affiliate program management */
class AffiliateResource(private val httpClient: HttpClient) {
    /** GET /api/affiliates/summary - Get Affiliate summary */
    fun getSummary(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/affiliates/summary", params)
    }

    /** GET /api/affiliates/campaigns - Get Affiliate campaigns */
    fun listCampaigns(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/affiliates/campaigns", params)
    }

    /** GET /api/affiliates/commissions - Get Affiliate commissions */
    fun listCommissions(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/affiliates/commissions", params)
    }

    /** GET /api/affiliates/payouts - Get Affiliate payouts */
    fun listPayouts(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/affiliates/payouts", params)
    }

    /** GET /api/affiliates/vouchers - Get Affiliate vouchers */
    fun listVouchers(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/affiliates/vouchers", params)
    }

    /** GET /api/affiliates/commissionplans - Get Affiliate commission plans */
    fun listCommissionPlans(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/affiliates/commissionplans", params)
    }

}

File: sdk/fsharp/src/HiTechCloudSDK/Affiliate.fs
namespace HiTechCloud.SDK

/// Affiliate: affiliate program management
type AffiliateResource(httpClient: HttpClient) =

    /// GET /api/affiliates/summary - Get Affiliate summary
    member this.GetSummary() =
        async {
            return! httpClient.getAsync("/api/affiliates/summary")
        }

    /// GET /api/affiliates/campaigns - Get Affiliate campaigns
    member this.ListCampaigns() =
        async {
            return! httpClient.getAsync("/api/affiliates/campaigns")
        }

    /// GET /api/affiliates/commissions - Get Affiliate commissions
    member this.ListCommissions() =
        async {
            return! httpClient.getAsync("/api/affiliates/commissions")
        }

    /// GET /api/affiliates/payouts - Get Affiliate payouts
    member this.ListPayouts() =
        async {
            return! httpClient.getAsync("/api/affiliates/payouts")
        }

    /// GET /api/affiliates/vouchers - Get Affiliate vouchers
    member this.ListVouchers() =
        async {
            return! httpClient.getAsync("/api/affiliates/vouchers")
        }

    /// GET /api/affiliates/commissionplans - Get Affiliate commission plans
    member this.ListCommissionPlans() =
        async {
            return! httpClient.getAsync("/api/affiliates/commissionplans")
        }

